joblib/testing.py

```diff
@@ -21,7 +21,7 @@
 assert_false = _dummy.assertFalse
 assert_equal = _dummy.assertEqual
 assert_not_equal = _dummy.assertNotEqual
-assert_raises = _dummy.assertRaises
+assert_raises = pytest.raises
```

Comment:
Is there an advantage of using pytest.raises in the tests affected by this PR?

For the record, my personal preference is to always use assert_raises_regex in order to check the error message. I have seen cases where the test pass but the exception is raised for a completely different reason.

pytest has some way with a context manager of checking the error message by the way.